brozzler/ydl.py

```diff
@@ -285,6 +285,7 @@ def ydl_postprocess_hook(d):
     ytdlp_url = page.redirect_url if page.redirect_url else page.url
     is_youtube_host = isyoutubehost(ytdlp_url)
     if is_youtube_host and YTDLP_PROXY:
+        MAX_YTDLP_ATTEMPTS = 4
```

Comment:
Redefining this here is very likely to be problematic. While Python doesn't strictly enforce this, `MAX_YTDLP_ATTEMPTS` is, semantically speaking, a constant. We shouldn't be redefining this again inline in a function. But furthermore, redefining this here for one URL being processed will redefine it for _every_ URL processed for the lifetime of the process because `MAX_YTDLP_ATTEMPTS` is defined in module scope.